indico/modules/events/registration/models/registrations.py

```diff
@@ -339,6 +339,11 @@ def _fill_from_registration():
     def has_files(self):
         return any(item.storage_file_id is not None for item in self.data)
 
+    @property
+    def sections_with_answered_fields(self):
+        return [x for x in self.registration_form.sections
+                if set(child.id for child in x.children) & self.data_by_field.viewkeys()]
```

Comment:
I believe `if any(child.id in self.data_by_field for child in x.children)` would be more efficient here since it stops after the first child that has been found (which is the first item in children in most cases) instead of building a set with all of them.